pages/dashboard/index.tsx

```diff
@@ -3,23 +3,27 @@ import { GetServerSideProps, GetServerSidePropsContext } from 'next'
 import { useState } from 'react'
```

Comment:
@vsauceda-cfa this is looking pretty good. The only two things I would say is that I'm seeing some slow down when changing info in the fields. Do you know why that might be?

Also, I think that debounce pattern might be better for translation than doing it on blur. This would mean that the translation would run after the user has stopped typing for a bit, usually a half second or so.

There's a good example of this in the `LiveDataSearch` component. 

Every time that `handleChange` is called, the value is put into a queue to be called in half a second if there are no more invocations of `handleChange`.

It's a bit confusing but I think it will seem more intuitive for the user. What do you think?

Reply:
Yea. That should be able to work. Might be a better approach